pkg/action/scan.go

```diff
@@ -191,51 +199,84 @@ func recursiveScan(ctx context.Context, c bincapz.Config) (*bincapz.Report, erro
 		logger.Debug("files found", slog.Any("path count", len(paths)), slog.Any("scanPath", scanPath))
 
 		// path refers to a real local path, not the requested scanPath
+		var wg sync.WaitGroup
```

Comment:
Concurrency changes start here and end on L305 (final construction of the `r.Files` map).